src/persist-client/src/read.rs

```diff
@@ -530,13 +517,35 @@ where
         mut self,
         as_of: Antichain<T>,
     ) -> Result<Subscribe<K, V, T, D>, Since<T>> {
-        let snapshot_batches = self.snapshot(as_of.clone()).await?.into();
-        let listen = self.listen(as_of).await?;
-        Ok(Subscribe::new(snapshot_batches, listen))
+        let snapshot_batches = self.snapshot(as_of.clone()).await?;
+        let listen = self.listen(as_of.clone()).await?;
+        Ok(Subscribe::new(snapshot_batches, as_of, listen))
+    }
+
+    fn lease_batch_parts(
+        &mut self,
+        batch: HollowBatch<T>,
+        metadata: LeasedBatchMetadata<T>,
+    ) -> impl Iterator<Item = LeasedBatch<T>> + '_ {
+        batch.keys.into_iter().map(move |key| {
+            LeasedBatch {
+                shard_id: self.machine.shard_id(),
+                reader_id: self.reader_id.clone(),
+                metadata: metadata.clone(),
+                batch: HollowBatch {
+                    desc: batch.desc.clone(),
+                    keys: vec![key],
+                    // This isn't quite right, but it's not used by anything
```

Comment:
nit: the sentence seems somewhat mangled

Reply:
Also, I hope we don't regret this one when someone _does_ try and use `len`. :sweat_smile:

Reply:
it disappears in a later commit in this PR!

Reply:
yeah, noticed this after :raised_hands: